client/src/components/ModalView.jsx

```diff
@@ -0,0 +1,85 @@
+// @flow
+
+import React from 'react';
+import { useState } from 'react';
+import { Modal, Button, Space, Input } from 'antd';
+import { ExclamationCircleOutlined } from '@ant-design/icons';
+
+type ModalProps = {
+  isAddCategory: boolean,
+  isDeleteCategory: boolean,
+};
```

Comment:
Something to help make this a bit cleaner - we can use Flow [union types as a way to represent an enum](https://flow.org/en/docs/types/literals/)!  So in this case, for example, we can just have some prop called `modalType: 'add' | 'rename' | 'delete'` (or something along those lines).  Then when we get to rendering, we can do a switch statement like in the doc I linked, or just use if statements (ex. `if modalType === 'add'`)